### ISSUE TITLE
[BUG] Leading question mark in SAS token causes issues

### ISSUE THREAD

**Issue by SukruthKS** [Azure/azure-sdk-for-java#6957]
**Describe the bug**
If there is a leading question mark in the SAS token provided to `BlobServiceClientBuilder`, it causes the question mark to appear twice in the request URL or in the middle of the query string.

***Exception or Stack Trace***
Request logs from running the below code snippet:

Case 1: Question mark appearing twice
```
INFO com.azure.storage.blob.implementation.BlobsImpl$BlobsService.getProperties - --> HEAD https://<redacted>.blob.core.windows.net/cont/blob.txt??sv=2019-02-02&ss=b&srt=sco&sp=rwdlac&se=2019-12-19T09:36:37Z&st=2019-12-19T01:36:37Z&spr=https&sig=<redacted>
```

Case 2: Question mark in the middle of the query string
```
INFO com.azure.storage.blob.implementation.ServicesImpl$ServicesService.submitBatch - --> POST https://<redacted>.blob.core.windows.net?comp=batch&?sv=2019-02-02&ss=b&srt=sco&sp=rwdlac&se=2019-12-19T09:36:37Z&st=2019-12-19T01:36:37Z&spr=https&sig=<redacted>
```
**To Reproduce**
Please see the below code snippet.

***Code Snippet***
```
String endpoint = "https://<redacted>.blob.core.windows.net/";

BlobServiceClient serviceClient = new BlobServiceClientBuilder()
    .endpoint(endpoint)
    .sasToken("?sv=2019-02-02&ss=b&srt=sco&sp=rwdlac&se=2019-12-19T09:36:37Z&st=2019-12-19T01:36:37Z&spr=https&sig=<redacted>")
    .httpLogOptions(BlobServiceClientBuilder.getDefaultHttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
    .configuration(Configuration.getGlobalConfiguration().put("AZURE_LOG_LEVEL", "1"))
    .buildClient();

try
{
    serviceClient.getBlobContainerClient("cont")
        .getBlobClient("blob.txt")
        .getProperties();
}
catch (BlobStorageException ex)
{
}

try
{
    BlobBatchClient batchClient = new BlobBatchClientBuilder(serviceClient).buildClient();
    BlobBatch batch = batchClient.getBlobBatch();
    batch.deleteBlob("cont", "blob.txt");
    batchClient.submitBatch(batch);
}
catch (BlobStorageException ex)
{
}
```

**Expected behavior**
SDK should be able to construct request URL correctly whether the SAS token contains a leading question mark or not.

**Screenshots**
NA

**Setup (please complete the following information):**
 - OS: Linux
 - IDE : IntelliJ
 - Version of the Library used: 12.0.0

**Additional context**
NA

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by gapra-msft** [comment]
Hi @SukruthKS, 

I believe this bug was fixed in this PR https://github.com/Azure/azure-sdk-for-java/pull/6168 and released in the `12.1.0` version of `azure-storage-blob`. Updating  to the latest version should help resolve your issue.

**Comment by SukruthKS** [comment]
@gapra-msft thanks! Tried with `12.1.0` and the issue is gone.
